### PR TITLE
[PREVIEW] Feature/sl 1789 duration for multi session listing

### PR DESCRIPTION
**Pull request by leszekgu**
Changes to the events related to the multipart approach

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/CreateListingRequestAction.java [hunk 1]
```diff
@@ -84,8 +87,12 @@ public FactMessage generateFactMessage() {
     public List<UserTransactionData> generateUserTransactionData() {
         List<UserTransactionData> userTransactionDataList = new ArrayList<>();
 
-        userTransactionDataList.add(prepareCreateUserTransactionData("hearingPart", hearingPart.getId(), 0));
-        userTransactionDataList.add(prepareCreateUserTransactionData("hearing", hearing.getId(), 1));
+        int orderNr;
+        for (orderNr = 0; orderNr < hearingParts.size(); orderNr++) {
```
Comment (r-kruk-kainos) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/actions/listingrequest/CreateListingRequestAction.java]: Why should we do that? it doesn't really mater if one of the hearing parts were created before or after another one. There could be put with the same order but probably before hearing. Did you try to revert changes on creation?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/HearingMapper.java [hunk 2]
```diff
@@ -10,17 +10,27 @@
...
 @Component
 public class HearingMapper {
-    public HearingPart mapToHearingPart(CreateHearingRequest createHearingRequest) {
-        HearingPart hearingPart = new HearingPart();
-        hearingPart.setId(UUID.randomUUID());
-        hearingPart.setHearingId(createHearingRequest.getId());
 
-        return hearingPart;
+    public List<HearingPart> mapToHearingParts(CreateHearingRequest createHearingRequest) {
+        //TODO: Think if this is needed, or BeanValidation is sufficient
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/HearingMapper.java]: yeah Min(1)

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java [hunk 3]
```diff
@@ -43,6 +43,7 @@ public HearingWithSessionsResponse(Hearing hearing) {
         this.reservedToJudge = hearing.getReservedJudge() != null ? hearing.getReservedJudge().getName() : null;
         this.sessions = hearing.getHearingParts()
             .stream()
+            //TODO: filter for null should be before and then safely ViewSessionResponse::new
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/model/response/HearingWithSessionsResponse.java]: ...

File: src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/HearingMapper.java [hunk 4]
```diff
@@ -10,17 +10,27 @@
...
+        //if (createHearingRequest.getNumberOfSessions() == 0) {
+        //throw new RuntimeException("New Hearing Should have NumberOfSession > 0");
+        //}
+        List<HearingPart> parts = new ArrayList<>();
+        for (int i = 0; i < createHearingRequest.getNumberOfSessions(); i++) {
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlevents/mappers/HearingMapper.java]: 👍 good classic code

**Pull request merged**